src/plugins/mmapstorage/dynarray.c

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file
+ *
+ * @brief Source for DynArray, a simple dynamic array for meta-key deduplication.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "dynarray.h"
+
+#include <kdblogger.h>
+
+/* -- DynArray Implementation ----------------------------------------------------------------------------------------------------------- */
```

Comment:
Please add a short description of what the dynarray does.

Reply:
Done.